src/transform.ts

```diff
@@ -56,24 +71,50 @@ export function processFileCRA(
   };
 }
 
+// TODO: We need to re-architect the CSS transform as follow
+// pre-processor (sass, stylus, less) => process(??) => post-processor (css modules, tailwindcss)
+// Reference to https://github.com/vitejs/vite/blob/c29613013ca1c6d9c77b97e2253ed1f07e40a544/packages/vite/src/node/plugins/css.ts#L652-L673
 export function processCss(src: string, filename: string): TransformedSource {
-  if (filename.endsWith('.module.css')) {
-    return processCSSModules(src, filename);
+  let cssSrc = src;
+  const isModule = cssModuleRE.test(filename);
+  const isPreProcessorFile = isPreProcessor(filename);
+
+  // Pure CSS
+  if (!isModule && !isPreProcessorFile && !isTailwindCSS(cssSrc)) {
+    const relativeFilename = getRelativeFilename(filename);
+    // Transform to a javascript module that load a <link rel="stylesheet"> tag to the page.
+    return {
+      code: `const relativeCssPath = "${relativeFilename}";
+  const link = document.createElement('link');
+  link.rel = 'stylesheet';
+  link.href = relativeCssPath;
+  document.head.appendChild(link);
+  
+  module.exports = JSON.stringify(relativeCssPath);`,
+    };
   }
-  if (filename.endsWith('.scss') || filename.endsWith('.sass')) {
-    return processSass(src, filename);
+
+  // Pre-processor (sass, stylus, less)
+  if (isPreProcessor(filename)) {
+    cssSrc = processSass(filename);
   }
 
-  const relativeFilename = getRelativeFilename(filename);
-  // Transform to a javascript module that load a <link rel="stylesheet"> tag to the page.
-  return {
-    code: `const relativeCssPath = "${relativeFilename}";
-const link = document.createElement('link');
-link.rel = 'stylesheet';
-link.href = relativeCssPath;
-document.head.appendChild(link);
+  // Post-processor
+  // CSS modules
+  if (isModule) {
+    return processCSSModules(cssSrc, filename);
+  }
+
+  // TailwindCSS
+  if (isTailwindCSS(cssSrc)) {
+    return processTailwindCSS(cssSrc, filename);
+  }
```

Comment:
I think it's better to dynamically update the config, then feed it into a single `postss` execution?